sycl/source/detail/device_impl.cpp

```diff
@@ -314,6 +314,10 @@ bool device_impl::has(aspect Aspect) const {
     return get_info<info::device::usm_restricted_shared_allocations>();
   case aspect::usm_system_allocations:
     return get_info<info::device::usm_system_allocations>();
+  case aspect::ext_intel_pci_device_id:
+    return getPlugin().call_nocheck<detail::PiApiKind::piDeviceGetInfo>(
+               MDevice, PI_DEVICE_INFO_PCI_DEVICE_ID, sizeof(pi_device_type),
+               &device_type, &return_size) == PI_SUCCESS;
```

Comment:
```suggestion
    return getPlugin().call_nocheck<detail::PiApiKind::piDeviceGetInfo>(
               MDevice, PI_DEVICE_INFO_PCI_DEVICE_ID, 0,
               nullptr, &return_size) == PI_SUCCESS;
```
The query has nothing to do with `pi_device_type`, so it is better to just ask for the return size of the query and see what it says.

Reply:
That is true but if you have a look at the other queries there, all of them use `device_type`. I tried to be consistent with existing code. I can change this but then I think all the the queries should be updated as well.

Reply:
I agree, it looks to be unintentional for the other cases and should be fixed. `ext_intel_pci_address` seems like it is doing the right thing.

I have opened a new issue for addressing the existing checks: https://github.com/intel/llvm/issues/7011

Reply:
Fixed it for `ext_intel_pci_device_id`.